src/main/cpp/imgUtils.cpp

```diff
@@ -0,0 +1,243 @@
+            -sin(radians), cos(radians), 0,
+            0, 0, 1
+    };
+    double* rot = rot_data;
+
+    // Combined transformation matrix
+    double m_data1[3*3];
+    cblas_dgemm(CblasRowMajor, CblasNoTrans, CblasNoTrans, 3, 3, 3, 1.0, t2, 3, rot, 3, 0.0, m_data1, 3);
+    double m_data2[3*3];
+    cblas_dgemm(CblasRowMajor, CblasNoTrans, CblasNoTrans, 3, 3, 3, 1.0, m_data1, 3, t1, 3, 0.0, m_data2, 3);
+    double* m = m_data2;
+    // Transform image
+    img_transform(img_in,rows,cols,rows,cols,m[0], m[1], m[2], m[3], m[4], m[5], fill_value, img_out);
+}
+
+double* imageCutout(double* img_in, int rows, int cols, int x, int y, int width, int height, double fill_value) {
+    // Allocate memory for the output image using MKL
+   double* img_out = new double[rows * cols];
+
+    if (width < 1 || height < 1) {
+        // Invalid width or height, return the input image as it is
+        cblas_dcopy(rows * cols, img_in, 1, img_out, 1);
+    } else {
+        int end_x = x + width - 1;
+        int end_y = y + height - 1;
+
+        int start_x = std::max(1, x);
+        int start_y = std::max(1, y);
+        end_x = std::min(cols, end_x);
+        end_y = std::min(rows, end_y);
+
+        // Copy the input image to the output image using MKL
+        cblas_dcopy(rows * cols, img_in, 1, img_out, 1);
+
+        // Fill the cutout region with the fill_value
+        for (int i = start_y - 1; i < end_y; ++i) {
+            for (int j = start_x - 1; j < end_x; ++j) {
+                img_out[i * cols + j] = fill_value;
+            }
+        }
+    }
+
+    return img_out;
+}
+
+double* imageCrop(double* img_in, int orig_w, int orig_h, int w, int h, int x_offset, int y_offset) {
+    // Allocate memory for the output image
+    double* img_out = new double[w * h];
+
+    int start_h = (std::ceil((orig_h - h) / 2)) + y_offset - 1 ;
+    int end_h = (start_h + h - 1);
+    int start_w = (std::ceil((orig_w - w) / 2)) + x_offset - 1;
+    int end_w = (start_w + w - 1);
+
+    // Create a mask to identify the cropped region
+    double* mask = new double[orig_w * orig_h];
+    double* temp_mask = new double[w * h];
+
+    // Set mask elements to 0 outside the cropped region and 1 inside
+    memset(mask, 0, orig_w * orig_h * sizeof(double));
+    for(int i = 0; i < h * w; i++) {
+     temp_mask[i] = 1;
+    }
+
+    for (int i = start_h; i <= end_h; ++i) {
+        for (int j = start_w; j <= end_w; ++j) {
+            mask[i * orig_w + j] = temp_mask[(i - start_h) * w + (j - start_w)];
+        }
+    }
+
+    // Apply the mask to crop the image
+    for (int i = 0; i < h; ++i) {
+        for (int j = 0; j < w; ++j) {
+            img_out[i * w + j] = img_in[(start_h + i) * orig_w + (start_w + j)] * mask[(start_h + i) * orig_w + (start_w + j)];
+        }
+    }
+
+    // Free memory for the mask
+    delete[] mask;
+    delete[] temp_mask;
+
+    return img_out;
+}
+
+void img_translate(double* img_in, double offset_x, double offset_y,
```

Comment:
cast the offsets in the first line rather than every call to offset_x and y, and maybe consider changing the API to use int offsets.